custom/icds_reports/utils/aggregation_helpers/distributed/agg_awc.py

```diff
@@ -429,6 +429,31 @@ def updates(self):
             'start_date': self.month_start
         }
 
+        yield """
+        UPDATE "{tablename}" agg_awc SET
+            app_version = ut.app_version,
+            commcare_version = ut.commcare_version
+        FROM (
+            SELECT distinct 
+                awc_id,
+                supervisor_id,
+                %(start_date)s::DATE AS month,
+                LAST_VALUE(app_version) over w as app_version,
+                LAST_VALUE(commcare_version) over w as commcare_version
+            FROM "{usage_table}" 
+            WHERE month = %(start_date)s WINDOW w as (
+                PARTITION BY awc_id, supervisor_id ORDER BY 
```

Comment:
is it possible to apply windows in the existing query above and let the `sum`s happen as they are happening right now and just find last_value for these two?  If that would keep the behavior of the existing query same then we should try that if that is not increasing the cost too much.

you can put that updated query here in the comment if above is true then I can try doing the benchmark before you doing the code change

Reply:
SELECT
            awc_id,
            supervisor_id,
            month,
            sum(pse) AS usage_num_pse,
            sum(gmp) AS usage_num_gmp,
            sum(thr) AS usage_num_thr,
            sum(add_household) AS usage_num_hh_reg,
            sum(add_person) AS usage_num_add_person,
            sum(add_pregnancy) AS usage_num_add_pregnancy,
            sum(home_visit) AS usage_num_home_visit,
            sum(bp_tri1) AS usage_num_bp_tri1,
            sum(bp_tri2) AS usage_num_bp_tri2,
            sum(bp_tri3) AS usage_num_bp_tri3,
            sum(pnc) AS usage_num_pnc,
            sum(ebf) AS usage_num_ebf,
            sum(cf) AS usage_num_cf,
            sum(delivery) AS usage_num_delivery,
            LAST_VALUE(app_version) over w as app_version,
            LAST_VALUE(commcare_version) over w as commcare_version,
            CASE WHEN (
                sum(due_list_ccs) + sum(due_list_child) + sum(pse) + sum(gmp) + sum(thr)
                + sum(home_visit) + sum(add_pregnancy) + sum(add_household)
            ) >= 15 THEN 1 ELSE 0 END AS usage_awc_num_active,
            sum(due_list_ccs) AS usage_num_due_list_ccs,
            sum(due_list_child) AS usage_num_due_list_child_health
        FROM "{usage_table}"
        WHERE month = %(start_date)s GROUP BY awc_id, month, supervisor_id, app_version, commcare_version, form_date WINDOW w as (
                PARTITION BY awc_id, supervisor_id ORDER BY 
                form_date RANGE BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING
            )

Reply:
I think this query will become very complex. I thnk lets go ahead with a separate query. Also, we don't need to apply the month filter on quering on usage ucr. we just have to pull the lastest form